src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -76,23 +81,87 @@ export class DatasetService {
             }
           : undefined,
       } as IndexPatternSpec;
-      const temporaryIndexPattern = await this.indexPatterns?.create(spec);
+      const temporaryIndexPattern = await this.indexPatterns?.create(spec, true);
       if (temporaryIndexPattern) {
         this.indexPatterns?.saveToCache(dataset.id, temporaryIndexPattern);
       }
     }
   }
 
-  public fetchOptions(
+  public async fetchOptions(
```

Comment:
Do we want to give an option to users to explicitly cache burst `fetchOptions` ?